demonstrations/tutorial_classical_kernels.py

```diff
@@ -0,0 +1,491 @@
+#. `Kernel-based training of quantum models with scikit-learn <https://pennylane.ai/qml/demos/tutorial_kernel_based_training.html>`_
+
+Now, a kernel for us is a real-valued function of two variables
+:math:`k(\cdot,\cdot)` from a given data domain :math:`x_1, x_2\in\mathcal{X}`.
+Further, we require a kernel to be symmetric to exchanging the variable
+positions :math:`k(x_1,x_2) = k(x_2,x_1)`.
+Finally, we will also want to enforce the kernels be positive semi-definite,
+but let's avoit getting lost in mathematical definitions, you can trust that
+all kernels featuring in this demo are positive semi-definite.
+
+If you take whichever textbook on kernel methods and search for the word
+``prominent", chances are you'll find it next to the word ``example" in a
+sentence that introduces the so-called Gaussian (Radial Basis Function) kernel
+:math:`k_\sigma`.
+For the sake of simplicity, we assume we are dealing with real numbers
+:math:`\mathcal{X}\subseteq\mathbb{R}`, in which case the Gaussian kernel looks
+like
+
+.. math:: k_\sigma(x_1, x_2) = e^{-\frac{\lvert x_1 - x_2\rvert^2}{\sigma}},
+
+where the variance :math:`\sigma` is a positive real tunable parameter (the
+generalization to higher-dimensional data is straightforward using the
+Euclidean norm :math:`\lVert x_1 - x_2 \rVert_2^2`.)
+Now for practical purposes the Gaussian kernel has the advantage of being
+simple enough to study as a function, while still yielding good performance for
+real-life tasks.
+
+In particular, one of the properties of the Gaussian kernel is that it is a
+shift-invariant function (also called stationary).
+That means that adding a constant shift to both arguments does not change the
+value of the kernel, that is for :math:`a\in\mathcal{X}`, we have
+:math:`k_\sigma(x_1 + a, x_2 + a) = k(x_1, x_2)`.
+At the same time, it also means we can express the kernel as a function of only
+one variable, the so-called lag (or shift) :math:`\delta = x_1 -
+x_2\in\mathbb{R}`, where we have
+
+.. math:: k_\sigma(x_1, x_2) = e^{-\frac{\lvert x_1 - x_2 \rvert^2}{\sigma}} =e^{-\frac{\delta^2}{\sigma}} = k_\sigma(\delta).
+
+Also, this combined with the property :math:`k(x_1, x_2) = k(x_2, x_1)` results
+in the new property :math:`k(\delta)=k(-\delta)`.
+
+Of course the Gaussian kenrel is not the only shift-invariant kernel out there.
+As it turns out, there are many others (REFERENCES OR POINTERS OR SOMETHING!)
+which are also used in practice and have the shift-invariance property.
+Nevertheless, here we will only look at the simple Gaussian kernel with
+:math:`\sigma = 1`:
+
+.. math:: k_1(\delta) = e^{-\delta^2},
+
+but all the arguments and code we use are also amenable to other kernels which
+fulfill the following mild restrictions:
+#. Shift-invariance.
+#. Normalization :math:`k(0)=1`.
+#. Smoothness (seen as quickly decaying Fourier spectrum).
+
+Let's warm up our coding by writing the simple Gaussian kernel!
+
+First, importers gonna import B-)
```

Comment:
```suggestion
First, importers gonna import B-).
```

I am not sure I get this one, but that should not be a criterion :)

Reply:
It's an adaption from "haters gonna hate", but it's probably one too many, I'll leave this comment open and rethink in next iteration whether we keep it or discard it